ipamutils/utils.go

```diff
@@ -48,3 +64,62 @@ func initGranularPredefinedNetworks() []*net.IPNet {
 	}
 	return pl
 }
+
+// InitAddressPools allows to initialize the local and global scope predefined
+// address pools to the desired values. It fails is invalid input is passed
+// or if the predefined pools were already initialized.
+func InitAddressPools(list []*PredefinedPools) error {
+	localPools := make([]*net.IPNet, 0, len(list))
+	//done := false
+
+	for _, p := range list {
+		if p == nil {
+			continue
+		}
+		_, b, err := net.ParseCIDR(p.Base)
+		if err != nil {
+			return fmt.Errorf("invalid base pool %q: %v", p.Base, err)
+		}
+		ones, _ := b.Mask.Size()
+		if p.Size <= 0 || p.Size < ones {
+			return fmt.Errorf("invalid pools size: %d", p.Size)
+		}
+		localPools = append(localPools, initPools(p.Size, b)...)
+	}
+	if localPools == nil {
+		localPools = initBroadPredefinedNetworks()
+	}
+	PredefinedBroadNetworks = localPools
+	return nil
+}
+
+func initPools(size int, base *net.IPNet) []*net.IPNet {
+	one, bits := base.Mask.Size()
+	mask := net.CIDRMask(size, bits)
+	n := 1 << uint(size-one)
```

Comment:
Do we support IPv6?  If so we should consider bounding the number of networks that we will allocate here, because one could easily specify some number in the billions or larger here.  Obviously this would be a config error, but would manifest as the system suddenly running out of memory and panicking.

Reply:
we dont support IPV6

Reply:
we actually do

Reply:
We do support IPv6 in bridge networks. But (if am not wrong), we dont have default subnets. The user must specify a `--subnet` option in order for `--ipv6` option to be useful. In that case, do we still think this is relevant for this PR ?